src/cpp/ext/otel/key_value_iterable.h

```diff
@@ -72,6 +77,20 @@ class KeyValueIterable : public opentelemetry::common::KeyValueIterable {
         }
       }
     }
+    if (OpenTelemetryPluginState().labels_injector != nullptr &&
+        !OpenTelemetryPluginState().labels_injector->AddOptionalLabels(
+            optional_labels_, callback)) {
+      return false;
+    }
+    if (active_plugin_options_view_ != nullptr) {
+      active_plugin_options_view_->ForEach(
```

Comment:
`ForEach` needs to be modified to stop iterating over the plugin options and return false if any of plugin option returns false.
After that, `ForEach` needs to return the false itself, so that we can replicate the logic that if any callback returned false, this function returns a false as well

Reply:
Done. PTAL. Thanks!